avr/boards.txt

```diff
@@ -241,6 +242,21 @@ menu.bootloader=Bootloader
 1284.menu.clock.1MHz_internal.build.f_cpu=1000000L
 
 
+# C++ Language Standard
+1284.menu.LanguageStandard.Default=Default
+1284.menu.LanguageStandard.Default.compiler.cpp.extra_flags=
```

Comment:
The `compiler.cpp.extra_flags` property is already in use by the LTO board option:
https://github.com/MCUdude/MightyCore/blob/640d21af5c6e077ab1c41726e72d1c507e294f26/avr/boards.txt#L114

Although it is possible to do this by using dedicated properties in each of the options and then defining `compiler.cpp.extra_flags` by referencing each of the dedicated properties (e.g., `compiler.cpp.extra_flags={lto_flags} {std_flag}`), the platform really shouldn't be using the `compiler.cpp.extra_flags` property at all, since it is intended to be left for the use of the platform user in order to allow them to customize the compilation command. If the platform authors uses these properties then when the user tries to use them their definition overrides the platform's definition of the property. The platform author has the power to use any arbitrary property name they like so they have no need to monopolize the user's properties.
Some related discussion on that subject here: https://github.com/arduino/arduino-cli/issues/846

Reply:
Good to know. I do not use LTO so I did not see the conflict / overwrite. I incorrectly assumed that `compiler.cpp.extra_flags` was unique to each menu item and was concatenated magically. After typing that previous sentence out, I realize how absurd that assumption was. 

Would updating the platform.txt to have a compiler.cpp.languageStandard variable make more sense? While specific to the current problem, it would never run afoul of other places in the build process.

Reply:
I think that's a good approach. The alternative would be to use a single property for all the board level extra flags for each compilation recipe. For example:
boards.txt
```
...
1284.menu.LTO.Os_flto.compiler.cpp.board_extra_flags.lto=-Wextra -flto -g
...
1284.menu.CppLanguageStandard.Cpp11.compiler.cpp.board_extra_flags.std=-std=gnu++11
...
1284.compiler.cpp.board_extra_flags={compiler.cpp.board_extra_flags.lto} {compiler.cpp.board_extra_flags.std}
...
```
platform.txt
```
compiler.cpp.flags=-c -g -Os {compiler.warning_flags} {compiler.cpp.board_extra_flags} -fno-exceptions -ffunction-sections -fdata-sections -fno-threadsafe-statics -MMD
```
But there's no functional different between the two approaches. It's purely a style choice, which would be for MCUdude to make. I'm only an interested 3rd party in this project.

Note that there is no magic to these particular property names. I just made something up at random for the sake of the example. They could just as well be named `foobar`. It can be a bit confusing because some property names have special treatment by the build system, but the platform author is also welcome to use arbitrary properties of any name they like as long as they don't collide with the build properties that have special treatment.

>I realize how absurd that assumption was.

I don't think so. The platform system is pretty complex and the fact that it uses this custom "properties" data format makes it even more difficult to understand. Fortunately, the documentation has improved a bit since the time I learned it mostly through reverse engineering the existing platforms and a ton of trial and error.

Reply:
I chose `compiler.cpp.languageStandard` over `compiler.cpp.board_extra_flags.std` because I have to deal with many compilers at work which do _not_ use `-std=` to specify the C++ language mode (--c++11, --eecpp, etc). Call it descriptive paranoia.

Regardless of the style chosen by MCUDude, I really want the ability to change the C++ language standard without having to manually modify platform.txt. Thank you for letting me know that I was clobbering options :).